FinTech.Logic/BusinessLogic/Authentication/AuthenticationBL.cs

```diff
@@ -0,0 +1,79 @@
+﻿using FinTech.Data;
+using System;
+using System.Linq;
+using FinTech.Common;
+
+namespace FinTech.Logic
+{
+    public class AuthenticationBL
+    {
+        public User Login(Guid userID, string strPassword)
+        {
+            if (userID == Guid.Empty)
+            {
+                throw new ArgumentException(ErrorMessages.InvalidUserID);
+            }
+
+            if (string.IsNullOrEmpty(strPassword) || strPassword.Length < 8)
+            {
+                throw new ArgumentException(ErrorMessages.InvalidLoginPassword);
+            }
+
+            var user = DataContext.Users.FirstOrDefault(u => u.UserID == userID && u.Password == strPassword);
+
+            if (user == null)
+            {
+                throw new ArgumentException(ErrorMessages.UserNotFound);
+            }
+
+            return user;
+        }
+
+        public bool Register(UserDTO userDto)
+        {
+            if (DataContext.Users.Any(u => u.Phone == userDto.Phone))
```

Comment:
Create a method for validation.
Do not include all in one place.

Register should only do the register work.

Reply:
Done